### PR TITLE
Add TDX measurement validation logic.

### PR DESCRIPTION
**Pull request by deeglaze**

### CODE REVIEW

File: gcetcbendorsement/cmd/tdx.go [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+	}
+	defer cleanup()
+	if c.textproto || (c.bytesform == gcetcbendorsement.BytesAuto && out.IsTerminal()) {
+		text, err := prototext.MarshalOptions{Multiline: true, Indent: "  "}.Marshal(policy)
+		if err != nil {
+			return fmt.Errorf("could not marshal policy as textproto: %v", err)
+		}
+		_, err = out.Write(text)
+		return err
+	}
```
[the remark this answers is not in the document]
Comment (deeglaze) [reply gcetcbendorsement/cmd/tdx.go:127]: The lint looks at control flow dominance and says to get rid of the else.

File: gcetcbendorsement/cmd/tdx.go [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+
+The mandatory PATH must be to a binary serialized VMLaunchEndorsement.
+`,
+		PersistentPreRunE: c.persistentPreRunE,
+		RunE:              c.runE,
+	}
+	cmd.Flags().StringVar(&c.out, "out", "-", "Path to output serialized check.Policy. "+
+		"Default - for stdout.")
+	cmd.Flags().StringVar(&c.outform, "outform", "auto", "One of textproto|bin|hex|base64|auto. "+
+		"Outputs the policy as either textproto or serialized binary proto in raw binary, or encoded "+
```
[the remark this answers is not in the document]
Comment (deeglaze) [reply gcetcbendorsement/cmd/tdx.go:149]: Fixed.

File: gcetcbendorsement/cmd/tdx.go [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+	return nil
+}
+
+func makeTdxCommand(ctx0 context.Context) *cobra.Command {
+	c := &tdxCommand{}
+	cmd := &cobra.Command{
+		Use: "tdx CMD [-base=PATH] [-overwrite] [-ram_gib=#]",
+		Long: `Outputs the extended go-tdx-guest check.Policy with endorsement reference values.
+
+The mandatory PATH must be to a binary serialized VMLaunchEndorsement.
```
[the remark this answers is not in the document]
Comment (deeglaze) [reply gcetcbendorsement/cmd/tdx.go:270]: base is not required, which is why it's in [option brackets]. the endorsement path is only mandatory in the policy subcommand, no? This is the parent command definition.

File: gcetcbendorsement/tdxpolicy.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+	var result *tcpb.Policy
+	if opts.Base == nil {
+		result = &tcpb.Policy{}
+	} else {
+		result = proto.Clone(opts.Base).(*tcpb.Policy)
+	}
+	var mrtds [][]byte
+	for _, m := range golden.Tdx.Measurements {
+		if opts.RAMGiB != 0 && m.GetRamGib() != uint32(opts.RAMGiB) {
```
[the remark this answers is not in the document]
Comment (deeglaze) [reply gcetcbendorsement/tdxpolicy.go:67]: Added.

File: gcetcbendorsement/cmd/tdx.go [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+	// Check -base flag
+	if c.base != "" {
+		baseContent, err := backend.IO.ReadFile(c.base)
+		if err != nil {
+			return fmt.Errorf("failed to read base policy file %q: %v", c.base, err)
+		}
+		c.basePolicy = &cpb.Policy{}
+		if err := proto.Unmarshal(baseContent, c.basePolicy); err != nil {
+			return fmt.Errorf("failed to unmarshal base policy file %q: %v", c.base, err)
+		}
```
[the remark this answers is not in the document]
Comment (deeglaze) [reply gcetcbendorsement/cmd/tdx.go]: Done.

File: gcetcbendorsement/cmd/tdx.go [hunk 6]
```diff
@@ -0,0 +1,286 @@
...
+	ramGiB    int
+	// derived
+	basePolicy *cpb.Policy
+}
+
+type tdxKeyType struct{}
+
+var tdxKey tdxKeyType
+
+type tdxPolicyCommand struct {
```
[the remark this answers is not in the document]
Comment (deeglaze) [reply gcetcbendorsement/cmd/tdx.go:47]: The documentation of the command is in the Use/Long fields in makeTdxCommand, no? The linter doesn't require documentation on unexported types, and I think these are self explanatory.

File: gcetcbendorsement/cmd/tdx.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (AdamCDunlap) [reply gcetcbendorsement/cmd/tdx.go:241]: It says "PATH" is mandatory, and the only use of "PATH" above is in "-base=PATH". If PATH refers to something else, then you should change the names.

[the remark this answers is not in the document]
Comment (deeglaze) [reply gcetcbendorsement/cmd/tdx.go]: ah I see it's copypasta in the wrong place.

**Review by AdamCDunlap — APPROVED** [review]

**Pull request merged**